service/service_test.go

```diff
@@ -214,17 +229,21 @@ func TestClose(t *testing.T) {
 				DoGetHealthCheckFunc: func(cfg *config.Config, buildTime string, gitCommit string, version string) (service.HealthChecker, error) {
 					return hcMock, nil
 				},
+				DoGetMongoDBFunc: func(ctx context.Context, cfg *config.Config) (service.DataStore, error) {
+					return mongoDBMock, nil
+				},
 			}
 
 			svcErrors := make(chan error, 1)
 			svcList := service.NewServiceList(initMock)
-			svc, err := service.Run(ctx, cfg, svcList, testBuildTime, testGitCommit, testVersion, svcErrors)
+			svc, err = service.Run(ctx, cfg, svcList, testBuildTime, testGitCommit, testVersion, svcErrors)
```

Comment:
Just for my own benefit: why is this change needed? Is that so that it reuses the existing `svc` and `err` variables? I think that if you revert these changes, you can then get rid of the top `var svc *service.Service`.

Reply:
in go lang the The short-declare operator ( := ) gives you a way to declare a variable and assign its value in a single statement, without needing to specify the type. But err was specified above  so the linter realises this early assignment correctly. Now we didn't redeclare and reassign err we have svc not declared. Therefore we have the svc declaration